che.sh

```diff
@@ -740,38 +740,66 @@ update_che_image() {
 
 execute_che_mount() {
   debug $FUNCNAME
-  if [ ! $# -eq 3 ]; then 
-    error "${CHE_MINI_PRODUCT_NAME} mount: Wrong number of arguments provided."
-    return
-  fi
 
-  MOUNT_PATH=$(get_mount_path "${2}")
+  # Determine the mount path to do the mount
+  info "${CHE_MINI_PRODUCT_NAME} mount: Setting local mount path to ${PWD}"
+  MOUNT_PATH=$(get_mount_path "${PWD}")
+  HOME_PATH=$(get_mount_path "${HOME}")
 
-  if [ ! -e "${MOUNT_PATH}" ]; then
-    error "${CHE_MINI_PRODUCT_NAME} mount: Path provided does not exist."
-    return
-  fi
+  # If extra parameter provided, then this is the port to connect to
+  if [ $# -eq 1 ]; then
+    info "${CHE_MINI_PRODUCT_NAME} mount: Connecting to remote workspace on port ${1}"
+    WS_PORT=${1}
 
-  if [ ! -d "${MOUNT_PATH}" ]; then
-    error "${CHE_MINI_PRODUCT_NAME} mount: Path provided is not a valid directory."
-    return
+  # Port not provided, let's do a simple discovery of running workspaces
+  else 
+    info "${CHE_MINI_PRODUCT_NAME} mount: Searching for running workspaces with open SSH port..."
+
+    CURRENT_WS_INSTANCES=$(docker ps -aq --filter "name=_che_dev-machine")
+    CURRENT_WS_COUNT=$(echo $CURRENT_WS_INSTANCES | wc -w)
+    
+    if [ $CURRENT_WS_COUNT -eq 0 ]; then
+      error "${CHE_MINI_PRODUCT_NAME} mount: We could not find any running workspaces"
+
+    elif [ $CURRENT_WS_COUNT -eq 1 ]; then
+      RUNNING_WS_PORT=$(docker inspect --format='{{ (index (index .NetworkSettings.Ports "22/tcp") 0).HostPort }}' ${CURRENT_WS_INSTANCES})
```

Comment:
I made a number of improvements to handle this:
1. I now get a list of all ws containers.
2. I added in a check to see if they have ssh or not.
3. I then print out a list of all the containers their SSH port & whether they have SSH.

```
INFO: che mount: Setting local mount path to /c/codenvy/tmp/dir 4
INFO: che mount: Searching for running workspaces with open SSH port...
INFO: che mount: Re-run with 'che mount <ssh-port>'
WS CONTAINER ID    HAS SSH?    SSH PORT
3ca5f2d5c302       y           32797
33c0ce87fabf       n
679494b6cae7       y           32788
```